tests/rptest/clients/rpk.py

```diff
@@ -62,8 +62,18 @@ class RpkTool:
     """
     Wrapper around rpk.
     """
-    def __init__(self, redpanda):
+    def __init__(self, redpanda, user=None, passwd=None):
         self._redpanda = redpanda
+        self._global_args = []
+
+        if user is not None and passwd is not None:
+            algorithm = 'SCRAM-SHA-256'
+            self.sasl_create_user(user, passwd, algorithm)
```

Comment:
It strikes me as unexpected behavior for the rpk client constructor to have user creation be a side effect. Should user creation be done explicitly where its required?